### PR TITLE
Bump mergo v0.3.8

### PR DESCRIPTION
**Pull request by zappy-shu**
**- What I did**

Bumps mergo from v0.3.7 to v0.3.8

full diff: imdario/mergo@v0.3.7...v0.3.8

includes:

* imdario/mergo#112 Add strict override
  * fixes imdario/mergo#111 WithOverride should be able to check types
* imdario/mergo#106 Fix merging of interface types with concrete values
* imdario/mergo#120 should not overwrite pointers directly, instead check embedded values
  * fixes imdario/mergo#114 Embedded struct of pointer types will overwrite the whole destination struct
* imdario/mergo#125 added WithOverrideEmptySlice config flag

**- How I did it**

Updated the vendor and added 2 special transforms to handle a change in how pointers in maps are handled in mergo: https://github.com/imdario/mergo/compare/v0.3.7...v0.3.8#diff-a103b5fee5fb31abebf12ee067c01da8R153

These transforms ensure that:
* ulimit config overrides correctly switch between single and soft/hard instead of merging the 2 together.
* service network config aliases are replaced with the override list instead of the lists being appended.

**- How to verify it**

The existing unit tests cover the special transforms as they have been put in place to preserve the existing functionality

**- Description for the changelog**

Bumps mergo dependency from v0.3.7 to v0.3.8

**- A picture of a cute animal (not mandatory but encouraged)**

cat-2019-11-19

### CODE REVIEW

**Comment by silvin-lubecki** [comment]
The linter is complaining 🦁 
```
cli/compose/loader/merge.go:206:49: mergeUlimitsConfig - result 0 (error) is always nil (unparam)
func mergeUlimitsConfig(dst, src reflect.Value) error {
                                                ^
cli/compose/loader/merge.go:213:56: mergeServiceNetworkConfig - result 0 (error) is always nil (unparam)
func mergeServiceNetworkConfig(dst, src reflect.Value) error {
```

**Comment by silvin-lubecki** [comment]
@thaJeztah PTAL

File: cli/compose/loader/merge_test.go [hunk 1]
```diff
@@ -1,8 +1,11 @@
 package loader
 
 import (
+	"reflect"
 	"testing"
 
+	"github.com/imdario/mergo"
+
```
Comment (thaJeztah) [inline cli/compose/loader/merge_test.go]: nit; should be grouped with the imports below

**Comment by thaJeztah** [comment]
all green; merging!

**Pull request merged**